zio-http/src/main/scala/zio/http/api/openapi/JsonRenderer.scala

```diff
@@ -0,0 +1,57 @@
+package zio.http.api.openapi
+
+import zio.http.api.Doc
+import zio.http.model.Status
+
+import java.net.URI
+import scala.collection.immutable.Iterable
+
+private[openapi] object JsonRenderer {
+
+  def renderFields(fieldsIt: (String, Any)*): String = {
+    if (fieldsIt.map(_._1).toSet.size != fieldsIt.size) {
+      throw new IllegalArgumentException("Duplicate field names")
+    } else {
+      val fields = fieldsIt
+        .filterNot(_._2 == None) // We don't render empty fields
+        .map { case (name, value) => s""""$name":${renderValue(value)}""" }
+      s"{${fields.mkString(",")}}"
+    }
+  }
+
+  private def renderValue(value: Any): String = value match {
+    case Some(value)                       => renderValue(value)
+    case o: OpenAPI                        => o.toJson
+    case s: Status                         => s.code.toString
+    case s: String                         => s""""$s""""
+    case n: Int                            => n.toString
+    case n: Double                         => n.toString
+    case n: Long                           => n.toString
+    case n: Float                          => n.toString
+    case d: Doc                            => renderDoc(d)
+    case b: Boolean                        => b.toString
+    case m: Map[_, _]                      => renderMap(m)
+    case l: Iterable[Any]                  => renderIterable(l)
+    case (k, v)                            => s"{${renderKey(k)}:${renderValue(v)}}"
+    case u: URI                            => s""""${u.toString}""""
+    case p: Product if p.productArity == 0 => renderSingleton(p)
+    case other                             => s""""${other.toString}""""
+  }
+
+  private def renderDoc(d: Doc) =
+    // todo render markdown
+    s""""${d.toPlaintext(color = false)}"""".replace("\n", "<br/>")
```

Comment:
In some cases, OpenAPI lets us embed CommonMark (or html), while other places, we must use plaintext. However, `renderValue` always uses plaintext?

Reply:
The plaintext rendering renders new lines. Which is not valid inside a json string, so I have to escape it.
The html rendering of doc, renders a full html doc, not just a snippet. Including a big block of styling. So it is not usable.

Reply:
Example HTML output for one link.

```
+{"doc":"<html><head><style>
      +        h1 {
      +            color: rgb(36, 41, 46);
      +            font-weight: 600;
      +            line-height: 1.25;
      +            margin-bottom: 16px;
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +        }
      +
      +        h2 {
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +            font-size: 24px;
      +            letter-spacing: 0px;
      +            word-spacing: 2px;
      +            color: rgb(36, 41, 46);
      +            font-weight: 600;
      +        }
      +
      +        h3 {
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +            font-size: 21px;
      +            letter-spacing: 0px;
      +            word-spacing: 2px;
      +            color: rgb(36, 41, 46);
      +            font-weight: 700;
      +        }
      +
      +        p {
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +            color: #24292e;
      +        }
      +
      +        .error {
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +            color: #24292e;
      +        }
      +
      +        a {
      +            border: 0;
      +            color: rgb(189, 39, 26);
      +            text-decoration: none;
      +            font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
      +            font-size: inherit;
      +            font-size: 100%;
      +            margin: 0;
      +            padding: 0;
      +            vertical-align: baseline;
      +        }
      +
      +        a:hover {
      +            color: rgb(0, 0, 0);
      +        }
      +
      +        pre {
      +          background-color: rgba(27, 31, 35, .05);
      +        }
      +
      +        code {
      +            border-radius: 3px;
      +            color: rgb(36, 41, 46);
      +            font-family: SFMono-Regular, Menlo, Monaco, Consolas, Liberation Mono, Courier New, monospace;
      +            font-size: 85%;
      +            margin: 0;
      +        }
      +    </style></head><body><p><a href="https://google.com">https://google.com</a></p></body></html>"}
```

Reply:
And this is also not a supported [CommanMark ](https://spec.commonmark.org/0.27/#html-blocks) html subset.

Reply:
Would you mind deleting all that unnecessary `<html>` and make it so it renders as inline HTML?